### PR TITLE
Move report option mapping into Report

### PR DESCRIPTION
**Pull request by mvz**
Follow-up to #551.

### CODE REVIEW

**Comment by troessner** [comment]
Please rebase.:)

**Comment by mvz** [comment]
Done.

File: lib/reek/report.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    # @api private
+    def self.warning_formatter_class(warning_format)
+      case warning_format
+      when :wiki_links
+        WikiLinkWarningFormatter
+      else
+        SimpleWarningFormatter
+      end
+    end
+  end
```
Comment (chastell) [inline lib/reek/report.rb]: Can we turn them all into `module_functions`? They’d be callable on `Report` but private if some class decides to `include Report`.

File: lib/reek/report.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      when :json
+        JSONReport
+      when :html
+        HTMLReport
+      when :xml
+        XMLReport
+      else # :text
+        TextReport
+      end
+    end
```
Comment (chastell) [inline lib/reek/report.rb]: I’m thinking whether a more compact syntax wouldn’t be nicer to read:

```Ruby
    def self.report_class(report_format)
      case report_format
      when :html then HTMLReport
      when :json then JSONReport
      when :text then TextReport
      when :xml  then XMLReport
      when :yaml then YAMLReport
      else fail "Unknown report format: #{report_format}"
      end
    end
```

Comment (troessner) [inline lib/reek/report.rb]: How about getting rid of quite some duplication by doing

```Ruby
REPORT_FORMAT_TO_CLASS = {
  html: HTMLReport
  # ...
}

def self.report_class(report_format)
  fail "Unknown report format: #{report_format}" unless REPORT_FORMAT_TO_CLASS.key?(report_format)
  REPORT_FORMAT_TO_CLASS[report_format]
end
```
?

File: lib/reek/report.rb (hunk 1 above)
Comment (troessner) [inline lib/reek/report.rb]: :+1: 
Let's kill "self".

File: lib/reek/report.rb (hunk 2 above)
Comment (chastell) [inline lib/reek/report.rb]: Right, I thought about this as an option as well, but didn’t want to disrupt the current solution too much. But if so then let’s use `Hash#fetch`, that’s exactly its role:

```Ruby
REPORT_CLASSES = {
  html: HTMLReport,
  # ...
}

module_function

def report_class(report_format)
  REPORT_CLASSES.fetch(report_format) do
    fail "Unknown report format: #{report_format}"
  end
end
```

Comment (troessner) [inline lib/reek/report.rb]: Good idea!

**Comment by troessner** [comment]
Can you please point that to master?

**Comment by mvz** [comment]
Sure.

**Comment by mvz** [comment]
I'll create a new PR once I've processed the comments.

**Pull request closed without merging**